Source/Core/DolphinQt2/FIFOWindow.cpp

```diff
@@ -0,0 +1,359 @@
+}
+
+void FIFOWindow::CreateWidgets()
+{
+  auto* layout = new QGridLayout;
+
+  // Info
+  auto* info_group = new QGroupBox(tr("File Info"));
+  auto* info_layout = new QHBoxLayout;
+
+  m_info_label = new QLabel;
+  info_layout->addWidget(m_info_label);
+  info_group->setLayout(info_layout);
+
+  info_group->setMinimumHeight(100);
+
+  // Object Range
+  auto* object_range_group = new QGroupBox(tr("Object Range"));
+  auto* object_range_layout = new QHBoxLayout;
+
+  m_object_range_from = new QSpinBox;
+  m_object_range_to = new QSpinBox;
+
+  m_object_range_from->setMinimum(0);
+  m_object_range_from->setMaximum(99998);
+  m_object_range_to->setMaximum(99999);
+  m_object_range_to->setValue(99999);
+
+  object_range_layout->addWidget(new QLabel(tr("From:")));
+  object_range_layout->addWidget(m_object_range_from);
+  object_range_layout->addWidget(new QLabel(tr("To:")));
+  object_range_layout->addWidget(m_object_range_to);
+  object_range_group->setLayout(object_range_layout);
+
+  // Frame Range
+  auto* frame_range_group = new QGroupBox(tr("Frame Range"));
+  auto* frame_range_layout = new QHBoxLayout;
+
+  m_frame_range_from = new QSpinBox;
+  m_frame_range_to = new QSpinBox;
+
+  m_frame_range_to->setMinimum(1);
+  m_frame_range_from->setMaximum(9);
+  m_frame_range_to->setMaximum(10);
+
+  frame_range_layout->addWidget(new QLabel(tr("From:")));
+  frame_range_layout->addWidget(m_frame_range_from);
+  frame_range_layout->addWidget(new QLabel(tr("To:")));
+  frame_range_layout->addWidget(m_frame_range_to);
+  frame_range_group->setLayout(frame_range_layout);
+
+  // Playback Options
+  auto* playback_group = new QGroupBox(tr("Playback Options"));
+  auto* playback_layout = new QGridLayout;
+  m_early_memory_updates = new QCheckBox(tr("Early Memory Updates"));
+
+  playback_layout->addWidget(object_range_group, 0, 0);
+  playback_layout->addWidget(frame_range_group, 0, 1);
+  playback_layout->addWidget(m_early_memory_updates, 1, 0);
+  playback_group->setLayout(playback_layout);
+
+  // Recording Options
+  auto* recording_group = new QGroupBox(tr("Recording Options"));
+  auto* recording_layout = new QHBoxLayout;
+  m_frame_record_count = new QSpinBox;
+
+  m_frame_record_count->setMinimum(1);
+  m_frame_record_count->setMaximum(3600);
+  m_frame_record_count->setValue(3);
+
+  recording_layout->addWidget(new QLabel(tr("Frames to Record:")));
+  recording_layout->addWidget(m_frame_record_count);
+  recording_group->setLayout(recording_layout);
+
+  // Action Buttons
+  m_load = new QPushButton(tr("Load..."));
+  m_save = new QPushButton(tr("Save..."));
+  m_record = new QPushButton(tr("Record"));
+
+  m_save->setEnabled(false);
+  m_record->setEnabled(false);
+
+  m_button_box = new QDialogButtonBox(QDialogButtonBox::Close);
+
+  layout->addWidget(info_group, 0, 0, 1, -1);
+  layout->addWidget(playback_group, 1, 0, 1, -1);
+  layout->addWidget(recording_group, 2, 0, 1, -1);
+  layout->addWidget(m_load, 3, 0);
```

Comment:
Hmm. These buttons each stretch out to a third of the width of the window, which looks weird. Maybe another QDialogButtonBox should be used here (with custom ActionRole buttons) to make the alignment appropriate for the platform?